### PR TITLE
fix limitrange flaky test

### LINKED ISSUE #120770: [Flaky Test] [sig-scheduling] LimitRange should create a LimitRange with defaults and ensure pod has those defaults applied. [Conformance]
##### Error text:
```
[FAILED] Pod "pfpod2" is invalid: [spec.containers[0].resources.requests: Invalid value: "600m": must be less than or equal to cpu limit of 500m, spec.containers[0].resources.requests: Invalid value: "600Gi": must be less than or equal to ephemeral-storage limit of 500Gi, spec.containers[0].resources.requests: Invalid value: "600Mi": must be less than or equal to memory limit of 500Mi]
In [It] at: test/e2e/scheduling/limit_range.go:227 @ 09/17/23 04:52:31.31

```
#### Recent failures:
[2023/9/17 12:35:11 ci-kubernetes-e2e-kubeadm-kinder-external-etcd-latest](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-kinder-external-etcd-latest/1703265761478316032)
[2023/9/13 20:32:03 ci-kubernetes-kind-e2e-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-kind-e2e-parallel/1701936237973082112)
[2023/9/12 06:16:56 ci-kubernetes-kind-ipv6-e2e-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-kind-ipv6-e2e-parallel/1701358676578668544)
[2023/9/6 21:28:07 ci-kubernetes-kind-e2e-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-kind-e2e-parallel/1699413609056047104)

/kind failing-test

/sig scheduling

This flakes before in https://github.com/kubernetes/kubernetes/issues/93782 and was fixed by #95807.
As this test is not flaky a lot, the priority may not be that high. But as this is a conformance test, we'd better to know why this would happen.

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind flake

#### What this PR does / why we need it:

```golang
// When the LimitRanger admission plugin find 0 items from the LimitRange informer cache,
// it will try to lookup LimitRanges from the local LiveLookupCache which liveTTL is 30s.
// If a LimitRange was deleted from the apiserver, informer watch the delete event and then
// handle it lead to the informer cache doesn't have any other items, but the local LiveLookupCache
// has it and not expired at the same time, the LimitRanger admission plugin will use the
// deleted LimitRange to validate the request. So the request will be rejected by the plugin
// till the item is expired.
//
// With the following retry, we can make sure the item is expired and the request will be
// validated as expected.
```

#### Which issue(s) this PR fixes:

Fixes #120770 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by carlory** [comment]
I'm not sure if it's bug in the LimitRanger admission plugin or it's expected behavior?

/sig api-machinery

**Comment by carlory** [comment]
/cc @liggitt 
can you take a look at it?

File: test/e2e/scheduling/limit_range.go [hunk 1]
```diff
@@ -223,7 +223,20 @@ var _ = SIGDescribe("LimitRange", func() {
...
 		ginkgo.By("Creating a Pod with more than former max resources")
 		pod = newTestPod(podName+"2", getResourceList("600m", "600Mi", "600Gi"), v1.ResourceList{})
-		_, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{})
+		// When the LimitRanger admission plugin find 0 items from the LimitRange informer cache,
+		// it will try to lookup LimitRanges from the local LiveLookupCache which liveTTL is 30s.
+		// If a LimitRange was deleted from the apiserver, informer watch the delete event and then
+		// handle it lead to the informer cache doesn't have any other items, but the local LiveLookupCache
+		// has it and not expired at the same time, the LimitRanger admission plugin will use the
+		// deleted LimitRange to validate the request. So the request will be rejected by the plugin
+		// till the item is expired.
```
Comment (Huang-Wei) [inline test/e2e/scheduling/limit_range.go:232]: Hrm... good to know that. Then the validation step "Verifying the LimitRange was deleted" is not sufficient.

**Review by Huang-Wei — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**